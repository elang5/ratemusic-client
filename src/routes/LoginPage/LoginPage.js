import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import './Login.css'

class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }
  
  render() {
    return (
      <div className="login-page">
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </div>
    )
  }
}

export default LoginPage
