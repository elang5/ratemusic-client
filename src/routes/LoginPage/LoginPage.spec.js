import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginPage from './LoginPage'

describe(`LoginPage Component`, () => {
  it(`renders the LoginPage component`, () => {
    const wrapper = shallow(<LoginPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})