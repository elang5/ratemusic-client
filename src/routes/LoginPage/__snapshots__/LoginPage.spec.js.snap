// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component renders the LoginPage component 1`] = `
<Fragment>
  <div
    className="login-page"
  >
    <LoginForm
      onLoginSuccess={[Function]}
    />
  </div>
</Fragment>
`;
